backend/app/main/service/family_service.py

```diff
@@ -1,68 +1,181 @@
 """Family Service"""
 from collections import defaultdict
+from sklearn.cluster import KMeans
+from sklearn.feature_extraction.text import TfidfVectorizer
 from app.main import db
 
 def get(relative, distance, year, citations):
     """List all relatives of a paper."""
+    distance, year, citations = correct_filters(distance, year, citations)
+    paper_query = create_paper_query(relative, distance, year, citations)
+    citations_query = create_citations_query(relative, distance, year, citations)
+    author_query = create_author_query(relative, distance, year, citations)
+    paper_dictionary = create_papers(paper_query)
+    paper_dictionary = add_citations(citations_query, paper_dictionary)
+    paper_dictionary = add_authors(author_query, paper_dictionary)
+    paper_list = paper_dictionary_to_paper_list(paper_dictionary)
+    sorted_paper_list = sort_paper_list(paper_list)
+    sorted_paper_list = add_clusters(sorted_paper_list)
+    return sorted_paper_list
+
+def correct_filters(distance, year, citations):
+    """If a filter is too big or too small, set it to a default value."""
     distance = int(distance)
-    distance = max(1, distance)
-    distance = min(3, distance)
+    distance = min(distance, 3)
+    distance = max(distance, 1)
+    year = int(year)
+    citations = int(citations)
+    citations = max(citations, 1)
+    return distance, year, citations
+
+def create_paper_query(relative, distance, year, citations):
+    """Get all papers from the requested family."""
+    paper_query = "" \
+                  "select distinct from_paper as paper, from_title as title, " \
```

Comment:
Just found something very useful for queries: Multi-line strings in Python! https://stackoverflow.com/a/10660443/4777124

Maybe for the next time ;)